dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/EventDataQueryRequest.java

```diff
@@ -354,6 +363,14 @@ public EventDataQueryRequestBuilder fromCriteria( EnrollmentAnalyticsQueryCriter
                 return builder.dimension( criteria.getDimension() );
         }
 
+        private String handleMultiOptions( String values, String timeField )
+        {
+            return Arrays.stream( withOptionSeparator( values ).split( OPTION_SEP ) )
+                .map( aValue -> String.join( DIMENSION_NAME_SEP,
+                    aValue, timeField ) )
+                .collect( Collectors.joining( OPTION_SEP ) );
+        }
+
```

Comment:
the core of this PR is here: basically thigs like `eventDate=202111;202201` were not properly parsed